watertap/core/zero_order_costing.py

```diff
@@ -3079,6 +3203,53 @@ def cost_peracetic_acid(blk):
             blk.unit_model.electricity[t0], "electricity"
         )
 
+    def cost_struvite_classifier(blk):
+        """
+        Method for costing struvite classifier unit.
+        """
+        t0 = blk.flowsheet().time.first()
+
+        # Get parameter dict from database
+        parameter_dict = blk.unit_model.config.database.get_unit_operation_parameters(
+            blk.unit_model._tech_type, subtype=blk.unit_model.config.process_subtype
+        )
+
+        # Get costing parameter sub-block for this technology
+        ref_state, HRT, size_cost = _get_tech_parameters(
+            blk,
+            parameter_dict,
+            blk.unit_model.config.process_subtype,
+            ["reference_state", "HRT", "sizing_cost"],
+        )
+
+        # Add cost variable and constraint
+        blk.capital_cost = pyo.Var(
+            initialize=1,
+            units=blk.config.flowsheet_costing_block.base_currency,
+            bounds=(0, None),
+            doc="Capital cost of unit operation",
+        )
+
+        expr = pyo.units.convert(
+            ref_state * HRT * size_cost,
+            to_units=blk.config.flowsheet_costing_block.base_currency,
+        )
```

Comment:
In the four new costing functions you have written, the costing expression is `ref_state * HRT * size_cost`, where `ref_state`, `HRT`, and `size_cost` are all constants specified in the YAML. So then the cost is a constant. Unless there is a reason for doing this that I am not seeing, I think that in all of these costing expressions you want to replace `ref_state` with the actual volumetric flowrate from the model, e.g.

    expr = pyo.units.convert(
        blk.unit_model.properties_in[t0].flow_vol * HRT * size_cost,
        to_units=blk.config.flowsheet_costing_block.base_currency,
    )

Then, the `reference_state` entry can be removed from all of the YAML files.

Reply:
Ok, that makes sense. So why do some of the models in zero_order_costing use these 3 parameters together. I was specifically looking at electrochemical_nutrient_removal since it has some similarities.

Reply:
@MarcusHolly I recall that @luohezhiming might have updated electrochemical_nutrient_removal by replacing state_ref with actual flowrate.

Reply:
Currently I am using the existing ZO model for the clarifier, which still uses reference_state. Should I leave that as is or get rid of the reference_state and add a function for clarifiers into zero_order_costing?

Reply:
It looks to me like electrochemical nutrient removal is the only model which is currently costed this way. I have no idea why it is that way. @adam-a-a, it looks to me like Chenyu has not updated it.

As for the ZO clarifier model, I don't see an associated costing method for it in `zero_order_costing.py`, so I'm not sure what to make of that....

Reply:
@adam-a-a  or @tarnold17 The process also has FeCl3 (a coagulant) being fed into the clarifier. So is it better to make a zero_order_costing function that includes FeCl3 for clarifiers (and remove the reference state from its yaml) or just leave things as they are and ignore the FeCl3 costs?

Reply:
Yea, they provide "lb FeCl3/MG", and I haven't included it as a component in the model. I'll be updating this PR as soon as I touch up on a few more things.